cargo-dist/src/tasks.rs

```diff
@@ -2428,17 +2434,112 @@ pub(crate) struct AnnouncementTag {
     pub rust_releases: Vec<(PackageIdx, Vec<String>)>,
 }
 
+/// details on what we're announcing (partially computed)
+struct PartialAnnouncementTag {
+    /// The full tag
+    pub tag: Option<String>,
+    /// The version we're announcing (if doing a unified version announcement)
+    pub version: Option<Version>,
+    /// The package we're announcing (if doing a single-package announcement)
+    pub package: Option<PackageIdx>,
+    /// whether we're prereleasing
+    pub prerelease: bool,
+}
+
 /// Parse the announcement tag and determine what we're announcing
-pub(crate) fn parse_tag(
+///
+/// `tag` being None here is equivalent to `--tag` not being passed, and tells us to infer
+/// the tag based on things like "every package has the same version, assume we're
+/// announcing that version".
+///
+/// `needs_coherent_announcement_tag = false` tells us to produce a result even if inference
+/// fails to find a tag that will unambiguously work. This is used by commands like `init`
+/// and `generate` which want to consider "everything" even if the user never actually
+/// could announce everything at once. In this case dummy values will appear in every field
+/// except for `AnnouncementTag::rust_releases` which will contain every distable binary
```

Comment:
should we rename `rust_releases` `distable_binaries`?

Reply:
Yeah I think this code is so old it was gesturing at the possibility of there being like, npm packages in the workspace that we understand and can announce releases of. I'm gonna avoid doing this rn for churn but will keep an eye on it for when we factor this stuff out to a module.